File: app/src/main/java/ua/dp/rename/dniprostreets/di/module/ComponentsInitializerModule.java
package ua.dp.rename.dniprostreets.di.module;

import dagger.Module;
import dagger.Provides;
import ua.dp.rename.dniprostreets.core.ComponentInitializer;
import ua.dp.rename.dniprostreets.core.LogInitializer;
import ua.dp.rename.dniprostreets.core.SnappyInitializer;
import ua.dp.rename.dniprostreets.repo.SnappyRepository;

@Module
public class ComponentsInitializerModule {

    @Provides(type = Provides.Type.SET)
    ComponentInitializer provideSnappyInitializer(SnappyRepository db) {
        return new SnappyInitializer(db);
    }

    @Provides(type = Provides.Type.SET)
    ComponentInitializer provideLogInitializer() {
        return new LogInitializer();
    }
}

File: app/src/main/java/ua/dp/rename/dniprostreets/di/module/ComponentsInitializerModule.kt
package ua.dp.rename.dniprostreets.di.module

import dagger.Module
import dagger.Provides
import dagger.multibindings.IntoSet
import ua.dp.rename.dniprostreets.core.ComponentInitializer
import ua.dp.rename.dniprostreets.core.LogInitializer
import ua.dp.rename.dniprostreets.core.SnappyInitializer
import ua.dp.rename.dniprostreets.repo.SnappyRepository

@Module
open class ComponentsInitializerModule {

    @Provides
    @IntoSet
    open fun provideSnappyInitializer(snappyRepository: SnappyRepository): ComponentInitializer {
        return SnappyInitializer(snappyRepository)
    }

    @Provides
    @IntoSet
    open fun provideLogInitializer(): ComponentInitializer {
        return LogInitializer()
    }
}
